common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -1819,6 +1820,10 @@ public static enum ConfVars {
     HIVE_STRICT_CHECKS_BUCKETING("hive.strict.checks.bucketing", true,
         "Enabling strict bucketing checks disallows the following:\n" +
         "  Load into bucketed tables."),
+    HIVE_STRICT_CHECKS_OFFSET_NO_ORDERBY("hive.strict.checks.offset.no.orderby", false,
+        "Enabling strict offset checks disallows the following:\n" +
+        "  OFFSET without ORDER BY.\n" +
+        "OFFSET is mostly meaningless when a result set doesn't have a total order."),
```

Comment:
I'm thinking about setting this true by default like the others. It would not be backward compatible but otherwise users may continue using it accidentally. Or printing a warning is enough? WDT?

Reply:
I personally agree with the idea as long as the incompatibility is introduced to the new major version. In my experience providing a managed Hive service, the usage of OFFSET without ORDER BY is not so high, and the ratio of mistakes in the small number of use cases is high. Warning is a more conservative approach, though.

Reply:
Ok. Please add the warning and I will approve this patch.